### ISSUE TITLE
Making mamba self-contained/Having user configured ~/.conda and ~/.condarc alternatives

### ISSUE THREAD

**Issue by amovsesyan** [mamba-org/mamba#3123]
# Troubleshooting docs
 My problem is not solved in the Troubleshooting docs
# How did you install Mamba?
Micromamba

# Search tried in issue tracker
self contained 

# Latest version of Mamba
 My problem is not solved with the latest version
# Tried in Conda?
Not applicable

# Describe your issue
I need to use python and environment/package managers across different clusters with a shared home directory. With this in mind, I need a tool that is truly self contained and doesn't use files/directories inside the home directory. I know that I can make a symlink for `~/.conda` to point somewhere else, but doing this means that I can't use micromamba on multiple clusters simultaneously.

## ~/.condarc
I've been able to override the use of the `~/.condarc` file by making a `.mombarc` file located in `${MAMBA_ROOT_PREFIX}/.mambarc` and removing the `.condarc` file in the home directory.
## ~/.conda/
Unfortunately I can't seem to find a similar solution to changing the location of where the `.conda` directory is located. Is there any solution to keep using a `.conda` directory with the `environments.txt` while specifying it be in another location (such as `${MAMBA_ROOT_PREFIX}/.conda/`) ?
## Solutions?
It would be really helpful if there was some `user_home` variable that could be configured to change where these files and directories are search for. Alternatively, being able to select an option to store them in `${MAMBA_ROOT_PREFIX}` would truly make mamba/micromamba self contained.

**Comment by amovsesyan** [comment]
Yes - I tried that! Unfortunately the issue isn't really in the rc file, since that can be moved and mamba will find it inside `${MAMBA_ROOT_PREFIX}/`. The issue is the more in the `~/.conda` folder, as it is always in the home directory, and it is automatically created whenever a new environment is created.

**Comment by AntoinePrv** [comment]
What do you see in the `~/.conda` folder ? I'm not super familiar with it.

**Comment by arthursw** [comment]
I am also trying to run micromamba in an  self-contained manner, and I thought about specifying the `--rc-file` for all of my `micromamba` commands. Unfortunately, when I run ` micromamba --rc-file /path/to/project/.mambarc activate envName`, I get the following error:

```
'micromamba' is running as a subprocess and can't modify the parent shell.
Thus you must initialize your shell before using activate and deactivate.

To initialize the current zsh shell, run:
    $ eval "$(micromamba shell hook --shell zsh)"
and then activate or deactivate with:
    $ micromamba activate
To automatically initialize all future (zsh) shells, run:
    $ micromamba shell init --shell zsh --root-prefix=~/.local/share/mamba
If your shell was already initialized, reinitialize your shell with:
    $ micromamba shell reinit --shell zsh
Otherwise, this may be an issue. In the meantime you can run commands. See:
    $ micromamba run --help

Supported shells are {bash, zsh, csh, posix, xonsh, cmd.exe, powershell, fish, nu}.
critical libmamba Shell not initialized
```

Is this a bug?

**Comment by arthursw** [comment]
Note that I did  initialize the current zsh shell with `eval "$(bin/micromamba --rc-file //path/to/project/.mambarc shell hook -s posix)"`.
